packages/storage/src/implementation/blob.ts

```diff
@@ -30,7 +30,7 @@ import * as type from './type';
  *     modified after this blob's construction.
  */
 export class FbsBlob {
-  private data_: Blob | Uint8Array;
+  private data_!: Blob | Uint8Array;
```

Comment:
Can you explain to me what the bang here means?

Reply:
It is a way of us promising the compiler that this variable will be initialized before anything calls it.  The compiler thinks data_ isn't definitively set in the constructor because there's an `if` and 2 `else if`s where it is set, but no `else`.  Since the constructor requires the `data` param to be of a type that would meet all possible `if` conditions, it should always get set in the constructor.

I can fix it by adding an else at the end that assigns `data_` some default value, which seems risky unless you have a suggestion, or I can just look at the types and promise the compiler `data_` is going to get set.  Thoughts?

Reply:
Interesting. I have never seen this before. We have a bunch of code in Firestore that we could simplify this way. So keep it as is please :)